introduce new intro subsection on SCONE Protocol Overview and Network Element Function (#63)
ietf-wg-scone/appman: draft-ietf-scone-applicability-manageability.md

---

> title: "Manageability Considerations for SCONE"

Author: ```suggestion
title: "Applicability & Manageability Considerations for SCONE"
```

area: "Web and Internet Transport"
workgroup: "Standard Communication with Network Elements"

> keyword: [SCONE, access networks, bitrate, throughput advice, manageability]

Author: ```suggestion
keyword: [SCONE, access networks, bit rate, throughput advice, applicability, manageability]
```

--- abstract

> This document describes manageability considerations for network operator for providing throughput guidance to

Author: ```suggestion
This document describes manageability considerations for network operators for providing throughput guidance to
```

> This document describes manageability considerations for network operators for providing throughput guidance to

Author: ```suggestion
This document describes the Applicability and Manageability considerations for network operators for providing throughput guidance to
```

application endpoints. This guidance is specifically addressed within the context of communication
networks utilizing the Standard Communication with Network Elements (SCONE) protocol.

--- middle

# Introduction

> The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum sustainable rate, to application endpoints via SCONE packets in the communication networks.

Author: ```suggestion
The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit to application endpoints via SCONE packets in the communication networks.
```

Network elements can provide notifications of the advisory maximum sustainable rate in each direction of the observed traffic. This allows applications, particularly those using adaptive bit-rate (ABR)
mechanisms, to proactively align their transmission rates with network policies. This document provides
operational, configuration, and management aspect for deploying the SCONE protocol within provider networks.

## SCONE Protocol Overview and Network Element Function

Deploying SCONE in an operator network involves the application
endpoints and any SCONE-capable network elements along the path of a
flow. SCONE is used on a flow only when its application endpoints
support it. SCONE packets use a QUIC header with a new version number
and are sent in the same UDP datagram as
the end-to-end QUIC packet using the same connection ID.
This ensures that network elements that forward QUIC packets also forward
the SCONE packets.
Specific network elements can
implement a SCONE network element function that sets

> the throughput advice.

Author: This section was moved up. I felt this is more like background on how scone work than providing any guidance in itself, therefore I propose this as a subsection in the introduction. That keeps the guidance section clear and focuses.

To participate in SCONE, a network element is assumed to have the
functional capability to identify and track SCONE-compliant QUIC
flows, recognize and process SCONE packets within those flows, and
map network policies into throughput advice to be inserted into the

> SCONE packets (see also {{network-integration}}.

Author: ```suggestion
SCONE packets.
```

By providing a standardized mechanism, SCONE allows network operators to provide
throughput advice information without custom APIs or per-network integrations. Applications can
self-adapt to the advised rate rather than relying on network rate limiters such as policers
or shapers, and the network can update the advised rate for an active flow, including to

> support tiered subscriber data plans (see {{Section 3.2 of I-D.ietf-scone-protocol}}).

Author: I moved this section up from the "In-band signalling and Network integration" section. This is a really nice paragraph explaining the benefits of scone, there I think that should be part of the general "background". There is no guidance here only explaining how the standard works.

--- abstract
This document describes the Applicability and Manageability considerations for network operators for providing throughput guidance to

> application endpoints. This guidance is specifically addressed within the context of telecommunication service

Author: ```suggestion
application endpoints. This guidance is specifically addressed within the context of telecommunications service
```

> networks utilizing the Standard Communication with Network Elements (SCONE) protocol.

Author: ```suggestion
application endpoints. This guidance is specifically addressed within the context of telecommunication service
provider networks utilizing the Standard Communication with Network Elements (SCONE) protocol.
```

--- middle

# Introduction

> The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit rate to application endpoints via SCONE packets in the telecommunications service provider networks.

Author: ```suggestion
The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit rate, to application endpoints via SCONE packets in the telecommunications service provider networks.
```

> The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit to application endpoints via SCONE packets in the communication networks.

Author: ```suggestion
The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit rate to application endpoints via SCONE packets in the telecommunications service provider networks.
```

--- abstract

> This document describes the Applicability and Manageability considerations for network operators for providing throughput guidance to

Author: ```suggestion
This document describes the Applicability and Manageability considerations for providing throughput guidance to
```

The SCONE protocol {{I-D.ietf-scone-protocol}} provides a signaling mechanism that enables on-path, SCONE-capable network elements to communicate "throughput advice", the advisory maximum allowable bit rate, to application endpoints via SCONE packets in the telecommunications service provider networks.

Network elements can provide notifications of the advisory maximum sustainable rate in each direction of the observed traffic. This allows applications, particularly those using adaptive bit-rate (ABR)
mechanisms, to proactively align their transmission rates with network policies. This document provides

> operational, configuration, and management aspect for deploying the SCONE protocol within provider networks.

Author: ```suggestion
Network elements capable of rate limiting can send notifications of the advisory maximum allowable bit rate in each direction of the observed traffic. This allows applications, particularly those using adaptive bit-rate (ABR)
mechanisms,to proactively align their transmission rates with network policies. This document addresses the
Applicability and Manageability considerations for deploying the SCONE protocol within service provider networks.
It also addresses operational, configuration, and management aspects not covered in the core protocol specification.
```

Applicability and Manageability considerations for deploying the SCONE protocol within service provider networks.
It also addresses operational, configuration, and management aspects not covered in the core protocol specification.

> ## SCONE Protocol Overview and Network Element Function

Reviewer: ```suggestion
## SCONE Protocol Overview and Network Element Function {#scone-overview}
```

Deploying SCONE in an operator network involves the application
endpoints and any SCONE-capable network elements along the path of a
flow. SCONE is used on a flow only when its application endpoints

> support it. SCONE packets use a QUIC header with a new version number

Reviewer: ```suggestion
Deploying SCONE in a communication network involves the application
endpoints and any SCONE-capable network elements along the path of a
flow of QUIC UDP datagrams. SCONE is used on a QUIC flow only when its application endpoints
support it. SCONE packets use a QUIC header with a new version number
```

Author: These changes make sense. Do we want to take them with this PR or separately?

Reviewer: One more small fix planned for this same area once this lands, the next sentence ("SCONE packets use a QUIC header with a new version number and are sent in the same UDP datagram as the end-to-end QUIC packet using the same connection ID") could tighten to name coalescing directly: "SCONE packets are QUIC long header packets carrying a SCONE-specific version number, coalesced into the same UDP datagram as the QUIC packets they accompany." Holding off posting it as a suggestion since it overlaps this one's last line, will add as a follow-up edit once this is applied.

Author: But should I merge this one now or we take everything together in a separate PR?

Reviewer: Yes, please go ahead and merge. After your merge, I have a proposed change that I held off because it applies to the same lines (82-84) that we touched previously. The proposed change:

From: "SCONE packets use a QUIC header with a new version number and are sent in the same UDP datagram as the end-to-end QUIC packet using the same connection ID."

To: "SCONE packets are QUIC long header packets carrying a SCONE-specific version number, coalesced into the same UDP datagram as the QUIC packets they accompany."

The reasoning for this change is it names coalescing directly, the actual mechanism described in Section 5 of the protocol spec, rather than describing it indirectly through "same UDP datagram... same connection ID."

Reviewer: Quick recap for #63, everything else is resolved: the new subsection heading anchor, the "SCONE-capable" wording, and both the "network types" and "QUIC endpoints/bitrate" paragraph fixes are all applied. This suggestion (lines 79-82) is the only one still open. Once you accept this, I'll add the last proposed change I outlined above (lines 82-84), and then the PR should be ready to merge.

and are sent in the same UDP datagram as

> the end-to-end QUIC packet using the same connection ID.

Reviewer: As promised, here's the follow-up for lines 82-84:
```suggestion
support it. SCONE packets are QUIC long header packets carrying a
SCONE-specific version number, coalesced into the same UDP datagram
as the QUIC packets they accompany.
```

This ensures that network elements that forward QUIC packets also forward
the SCONE packets.
Specific network elements can
implement a SCONE network element function that sets

> the throughput advice.

Reviewer: Good catch, that is circular since this sentence defines what makes a network element SCONE-capable. Simpler fix:
```suggestion
Any network element can
implement the SCONE network element function that sets
the throughput advice.
```

Author: I think using "SCONE-capable" here is superficial, as this sentence is basically explain what a SCONE-capable network element is - an element that implement the SCONE network function.

To participate in SCONE, a network element is assumed to have the
functional capability to identify and track SCONE-compliant QUIC
flows, recognize and process SCONE packets within those flows, and
map network policies into throughput advice to be inserted into the
SCONE packets.

> By providing a standardized mechanism, SCONE allows network operators to provide
> throughput advice information without custom APIs or per-network integrations. Applications can
> self-adapt to the advised rate rather than relying on network rate limiters such as policers
> or shapers, and the network can update the advised rate for an active flow, including to
> support tiered subscriber data plans (see {{Section 3.2 of I-D.ietf-scone-protocol}}).

Reviewer: ```suggestion
By providing a standardized mechanism, SCONE allows network operators to provide
throughput advice information to QUIC endpoints without custom APIs or per-network integrations. Applications can
self-adapt to the advised bitrate rather than relying on network rate limiters such as policers
or shapers, and the network can update the advised bitrate for an active flow, including to
support tiered subscriber data plans (see {{Section 3.2 of I-D.ietf-scone-protocol}}).
```

Author: I guess I didn't undo this change from bit-rate to rate. Again I was intending to not do wording changes in this PR but only move things. But I guess this is a small enough change that we can take it now.

Reviewer: Yes, let's take this now.

Author: I merged it but it might create a merge conflict with #79. However, I can fix that later in that PR...

such as wireline broadband or Wi-Fi, may apply policies at centralized aggregation points
or gateways such as the Broadband Network Gateway serving multiple devices.

Encompassing deployment of network elements in a wide range of networks, this document
is limited to discussing the core Applicability and Manageability considerations for

> the SCONE protocol to ensure its consistent and effective use across varied network paths.

Author: This was moved down, however, I'm actually not sure what the value of being consistent across networks is. Is that actually the goal? Why? I thought it more about providing advice to do it "right".

Author: I opened a separate issue for this. See #64

Author: Just to add: I moved this down because this sentence explain the goal and scope of the guidance in the next section, therefore I thought this is the perfect sentence to start the next section in the preamble before the subsection with each specific guidance.

> # Manageability Considerations

Author: Should we say "Considerations" or "Guidance" or both?

```suggestion
# Manageability Considerations and Guidance
```

Author: ```suggestion
# Applicability, Manageability and Operational Considerations
```

# Terms and Definitions

This document uses terms and definitions described in {{I-D.ietf-scone-protocol}}.

# Applicability, Manageability and Operational Considerations
# Manageability Considerations

Deploying SCONE in an operator network involves the application
endpoints and any SCONE-capable network elements along the path of a
flow. SCONE is used on a flow only when its application endpoints
support it. Network elements that forward QUIC packets also forward
the SCONE packets carried among them, and specific network elements
implement and configure the SCONE Network Element function that sets
the throughput advice. This document as a whole covers the
applicability, manageability, and operational considerations for

> deploying SCONE in such networks.

Author: This was moved up in a new overview subsection in the intro as it is more background information that guidance/consideration.

> Because throughput advice applies strictly to this specific flow, SCONE network elements

Author: ```suggestion
Because throughput advice applies strictly to this specific flow, SCONE Network Elements
```

While the signaling itself is stateless, managing the operational lifecycle of a SCONE
deployment may require establishing and maintaining per-flow context. Specifically, to execute
the monitoring, logging, and conformance evaluation functions detailed later in this document
(see {{conformance-monitoring}}),

> the network element has to track the flow's throughput over multiple monitoring periods. This

Author: This is unfortunately not displayed nicely, but here I only changed "requires" to "may require". Because as the document say in the previous paragraph, it is not always required. This is an editorial correction.

Then I also added the reference to the conformance monitoring section.

such as wireline broadband or Wi-Fi, may apply policies at centralized aggregation points
or gateways such as the Broadband Network Gateway serving multiple devices.

# Terms and Definitions

This document uses terms and definitions described in {{I-D.ietf-scone-protocol}}.

# Applicability, Manageability and Operational Considerations

Encompassing deployment of network elements in a wide range of networks, this document
is limited to discussing the core manageability considerations for

> the SCONE protocol to ensure its consistent and effective use across varied network paths.

Reviewer: ```suggestion
Encompassing deployment of network elements in a wide range of networks, this document
is limited to discussing the core manageability and operational considerations for
the SCONE protocol to support its effective use across these varied network types.
```

Author: As I wrote as a comment I only moved the text and didn't edit it. I thought we would then later take these edits in a separate PR but we can also do it now...?

Reviewer: Yes, let's take this now too, rather than a separate PR.

Author: I merged it but it might create a merge conflict with my proposed text in PR #77. But I can fix that later in #77.

per-flow state.

While the signaling itself is stateless, managing the operational lifecycle of a SCONE
deployment may require establishing and maintaining per-flow context. Specifically, to execute
the monitoring, logging, and conformance evaluation functions detailed later in this document

> (see {{conformance-monitoring}}),

Author: ```suggestion
the monitoring, logging, and conformance evaluation functions detailed later in this document,
```

## Presence of SCONE Network Elements On the Data Path
When multiple SCONE-capable network elements are present on the same data path, they operate
independently, with no synchronization or control-plane coordination required between them.

> This is possible because each network element is expected to only lowers the rate signal,

Author: This is an editorial rephrasing that makes it clear that this is the expected behaviour but there is no guarantee for it.

This is possible because each network element is expected to only lowers the rate signal,
preserving any lower advice already set by
another element on the path, so the endpoint applies the most restrictive advice along the
path (see {{Section 5.4 of I-D.ietf-scone-protocol}}). This lets operators deploy and manage

> SCONE network elements independently, without building integration between them.

Author: This sentence is removed because it was redundant.

path (see {{Section 5.4 of I-D.ietf-scone-protocol}}).

## Change of Network Element During an Active Flow
The on-path network element can change when an application changes its access network, for
example during QUIC connection migration or a mobility event where the IP address is unchanged.
Because SCONE signaling is stateless, this transition needs no explicit teardown or state
transfer between the old and new network elements. The endpoint and network elements follow the

> migration steps defined in {{Section 6.3 of I-D.ietf-scone-protocol}}, where the endpoint sends

Author: Removed "migration steps" because while the section in the spec talks about migration, it doesn't define a set of steps, therefore I found this confusing.

guidance in {{Section 7.2 of I-D.ietf-scone-protocol}}.

> ## Conformance Monitoring {#conformance-monitoring}

Author: ```suggestion
## Conformance Monitoring
```

## Conformance Monitoring {#conformance-monitoring}
Networks that choose to provide SCONE throughput advice can implement mechanisms to

> monitor QUIC flows and measure conformance to the advised bit-rate, either per flow of

Author: We should double-check the terminology. The scone protocol spec purposefully mostly talks about throughput advice. Would be good to align the terminology here as well.

I will open an issue for this.

Author: Actually this change from bit-rate to rate, doesn't make a but difference. However, the protocol draft used "bitrate" (without the dash). As I was editing it, I just thought rate is slightly better. however, maybe we should rather say "throughput advice".

> monitor QUIC flows and measure conformance to the advised rate, either per flow of

Author: ```suggestion
monitor QUIC flows and measure conformance to the advised bit-rate, either per flow of
```

contribute to a shared policy limit (such as a device or network subscription level). This

> will allow operators to validate whether applications are following the advised rate.

Author: ```suggestion
will allow operators to validate whether applications are following the advised bit-rate.
```

> While it is expected that operators will implement monitoring at the SCONE network element

Author: ```suggestion
While it is expected that operators will implement monitoring at the SCONE Network Element
```

providing the advice, it could also be performed elsewhere in the network. However, network
elements lack the capability to validate the legitimacy of SCONE packets coalesced with other
QUIC packets. Therefore, operators must ensure a network element evaluates conformance only

> against the advised rate that it set itself, and never enforces limits based on advice

Author: ```suggestion
against the advised bit-rate that it set itself, and never enforces limits based on advice
```

can deploy SCONE purely as an advisory signal without any throttling fallback, prioritizing
cooperative application optimization over strict compliance enforcement.

> ## Network Integration {#network-integration}

Author: ```suggestion
## In-Band Signaling and Network Integration
```

can deploy SCONE purely as an advisory signal without any throttling fallback, prioritizing
cooperative application optimization over strict compliance enforcement.

## In-Band Signaling and Network Integration
## Network Integration {#network-integration}

Because SCONE packets are always coalesced with ordinary QUIC packets, SCONE signaling
operates entirely in-band. It does not introduce any additional routing overhead or

> require the creation of out-of-band signaling interfaces. Instead, SCONE signaling

Author: This sentence is merged into the new last sentence below as it was a bit redundant.

operates entirely in-band. SCONE signaling
inherently traverses the already established network path, such as the existing
connection between a user device and a network gateway, associated with the QUIC flow
for which the network element intends to send throughput advice. This ensures that
SCONE seamlessly integrates into existing architectures without requiring new tunnels
or data paths to be established.

By providing a standardized mechanism, SCONE allows network operators and QUIC endpoints to
exchange bit-rate information without custom APIs or per-network integrations. Applications can
self-adapt to the advised bit-rate rather than relying on network rate limiters such as policers
or shapers, and the network can update the advised bit-rate for an active flow, including to

> support tiered subscriber data plans (see {{Section 3.2 of I-D.ietf-scone-protocol}}).

Author: This paragraph was moved into the intro/overview section. Maybe we should move there rest as well? I added a reference already. In any case we could maybe move this section early in the document.

cooperative application optimization over strict compliance enforcement.

## In-Band Signaling and Network Integration

Because SCONE packets are always coalesced with ordinary QUIC packets, SCONE signaling
operates entirely in-band. SCONE signaling
inherently traverses the already established network path, such as the existing
connection between a user device and a network gateway, associated with the QUIC flow

> for which the network element intends to send throughput advice. This ensures that
> SCONE seamlessly integrates into existing architectures without requiring new tunnels
> or data paths to be established.

Author: ```suggestion
for which the network element intends to send throughput advice. As such,
SCONE seamlessly integrates into existing architectures without requiring
new data paths, e.g. using tunnels, nor introducing any additional routing overhead.
Similarly, operators are not required to deploy any additional out-of-band signaling interfaces.
```

cooperative application optimization over strict compliance enforcement.

## In-Band Signaling and Network Integration

Because SCONE packets are always coalesced with ordinary QUIC packets, SCONE signaling
operates entirely in-band. SCONE signaling
inherently traverses the already established network path, such as the existing
connection between a user device and a network gateway, associated with the QUIC flow
for which the network element intends to send throughput advice.
As such, SCONE seamlessly integrates into existing architectures without requiring
new data paths, e.g. using tunnels, nor introducing any additional routing overhead.

> Similarly, operators are not required to deploy any additional out-of-band signaling interfaces.

Author: ```suggestion
for which the network element intends to send throughput advice. This ensures that
SCONE seamlessly integrates into existing architectures without requiring new tunnels
or data paths to be established.
```
